### PR TITLE
ManualAddSearchEngineSettingsFragment: Add 'final' keyword to constan…

### PR DESCRIPTION
**Pull request by michaelsam94**
…t #1972

### CODE REVIEW

**Comment by mcomella** [comment]
Hey @michaelsam94 – thanks for the comments and updated commit! We really appreciate it! Sorry this review has been taking so long – we're having a colocated team meeting and our schedule has been a bit crazy. I'll get to this review either later today or tomorrow morning.

**Comment by michaelsam94** [comment]
@mcomella never mind I really happy to contribute in this project thanks.

File: app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java [hunk 1]
```diff
@@ -72,9 +72,9 @@ public static SettingsFragment newInstance(Bundle intentArgs, SettingsScreen set
             case REMOVE_ENGINES:
                 f = new SettingsFragment();
                 break;
-            case ADD_SEARCH:
+            /*case ADD_SEARCH:
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java]: When removing code, you can just delete it – no need to comment it out! If we need to revive the code, we can get it from version control.

That being said, I'm not convinced we should delete this – see below.

File: app/src/main/java/org/mozilla/focus/settings/ValidateSearchEngineAsyncTask.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+
+
+
+public class ValidateSearchEngineAsyncTask extends AsyncTask<Void, Void, Boolean> {
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/settings/ValidateSearchEngineAsyncTask.java]: I don't think it's worth removing this from `Manual...Fragment` – we had to change the visibility of two mutable variables, which is already a really fragile pattern, so I don't want to expose them to other classes. The first alternative I think of, to use a listener to change the values, is a lot of boilerplate. I would prefer to just leave this as a private inner class of `Manual...Fragment`.

It's true that Manual... becomes a really long file and that this validation logic is a separate responsibility from managing the settings UI (so it does make sense to separate) but I think it'd be better to try to fix it by:
- Using more expressive constructs than AsyncTasks, e.g. kotlin coroutines.
- Taking *all* of the UI logic away from the validate async task
- Separating the validation logic from the async-handling logic as much as possible

Note that this async task was written last minute so it is a bit of a mess. :\

File: app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java [hunk 3]
```diff
@@ -7,53 +7,41 @@
...
-
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-
-public class ManualAddSearchEngineSettingsFragment extends SettingsFragment {
+public class ManualAddSearchEngineSettingsFragment extends PreferenceFragment {
     private static String LOGTAG = "ManualAddSearchEngine";
 
-    // Set so the user doesn't have to wait *too* long. It's used twice: once for connecting and once for reading.
-    private static int SEARCH_QUERY_VALIDATION_TIMEOUT_MILLIS = 4000;
+
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java]: nit: there should only be one line of whitespace between any part of the code.

File: app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java [hunk 4]
```diff
@@ -135,107 +123,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
     }
 
-    private void setUiIsValidatingAsync(final boolean isValidating, final MenuItem saveMenuItem) {
+    void setUiIsValidatingAsync(final boolean isValidating, final MenuItem saveMenuItem) {
         final ManualAddSearchEnginePreference pref =
                 (ManualAddSearchEnginePreference) findPreference(getString(R.string.pref_key_manual_add_search_engine));
         pref.setProgressViewShown(isValidating);
 
         saveMenuItem.setEnabled(!isValidating);
     }
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java]: nit: there's a lot of newline whitespace at the end of the file

File: app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java [hunk 5]
```diff
@@ -194,7 +194,10 @@ public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preferen
             showSettingsFragment(SettingsScreen.SEARCH_ENGINES);
             TelemetryWrapper.openSearchSettingsEvent();
         } else if (preference.getKey().equals(resources.getString(R.string.pref_key_manual_add_search_engine))) {
-            showSettingsFragment(SettingsScreen.ADD_SEARCH);
+            getFragmentManager().beginTransaction()
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java]: This statement is equivalent to the one you removed but less high-level – did you have a particular reason for changing the code? I find the high-level `showSettingsFragment(SettingsScreen.ADD_SEARCH)` to be clearer and it lets our SettingsScreen enum tell us all the different search screens we can show (if it's kept up to date).

**Comment by mcomella** [comment]
Just an FYI that #1972 was closed but #1945 is still open for completion.

**Comment by mcomella** [comment]
This PR seems inactive: closing. @michaelsam94 please let us know if you're still working on this and we can reopen.

**Pull request closed without merging**